assets/app/views/builds.html

```diff
@@ -138,6 +152,20 @@ <h2 ng-if="buildConfigName != ''">
               <span ng-switch-default class="fa fa-refresh fa-spin" aria-hidden="true"></span>
             </span>
           </div>
+          <div>
+            <span style="margin-right: 5px;">Duration: </span>
+            <span ng-switch="build.status" class="hide-ng-leave">
+              <span ng-switch-when="Complete">{{build.startTimestamp | duration : build.completionTimestamp}}</span>
+              <span ng-switch-when="Failed">{{build.startTimestamp | duration : build.completionTimestamp}}</span>
+              <span ng-switch-when="Running">running for <duration-till-now timestamp="build.startTimestamp"></duration-till-now></span>
+              <span ng-switch-when="New">waiting for <duration-till-now timestamp="build.metadata.creationTimestamp"></duration-till-now></span>
+              <span ng-switch-when="Pending">waiting for <duration-till-now timestamp="build.metadata.creationTimestamp"></duration-till-now></span>
+              <span ng-switch-default>
+                <span ng-if="build.startTimestamp">{{build.startTimestamp | duration : build.completionTimestamp}}</span>
```

Comment:
when duration gets an undefined / null value for the second value it falls back to assume it should be compared against the current time, the only reason for startTimestamp to exist but not completionTimestamp would be if the build is still in progress, theoretically that means the state should be in Running but threw this is in as a default in case someone changes the states without us realizing it.